fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -411,6 +416,24 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         }
     }
 
+    private fun maybeShowSplashScreen() {
+        if (components.settings.isFirstSplashScreenShown) {
+            return
+        } else {
+            components.settings.isFirstSplashScreenShown = true
+            // Splash screen compat fails to draw icons on earlier versions.
+            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.M) return
+        }
+
+        if (FxNimbus.features.splashScreen.value().enabled) {
+            val splashScreen = installSplashScreen()
+            var keepSplashOnScreen = true
+            val delay = FxNimbus.features.splashScreen.value().maximumDurationMs.toLong()
+            splashScreen.setKeepOnScreenCondition { keepSplashOnScreen }
+            Handler(Looper.getMainLooper()).postDelayed({ keepSplashOnScreen = false }, delay)
```

Comment:
Was there a reason to use a Handler over the `MainScope`? We don't usually use them directly.

Reply:
Honestly, I just didn't know we have the preferred way and used `Handler.postDelayed` after I saw 10+ usages across the project. Changed to `MainScope(delay = `👍

Reply:
Some times they are valid use cases, some times they are old code that we haven't touched in a long time. 🥲